## 資料的基本讀取



df.head() #展示資料前5行
df.tail() #展示資料後5行
df.head(8) #展示資料前8行
df.tail(17) #展示資料後17行


In [70]:
#引用 pandas套件並縮寫為pd
import pandas as pd
#資料表是利用pd.read_csv 讀取資料所生成
df = pd.read_csv("/home/kesci/input/china9852/中国各省 - simple.csv")
#展示資料前5列
df.head()

行政区  地区  Area(km2)  GDP(CNY 十亿)  人口(千人)  211大学  5A 景区   rain  TopMountains
0  重庆  西南      82374          1950   30325      2      8  1,439          2797
1  云南  西南     383186          1653   47562      1      8  1,100          6740
2  西藏  西南    1202189           131    3273      1      2    400          8844
3  四川  西南     486116          3698   82330      5     12  1,100          7556
4  贵州  西南     176099          1354   35425      1      5  1,200          2900

In [71]:
#資料的基本資訊
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 31 entries, 0 to 30
Data columns (total 9 columns):
行政区             31 non-null object
地区              31 non-null object
Area(km2)       31 non-null int64
GDP(CNY 十亿)    31 non-null int64
人口(千人)          31 non-null int64
211大学           31 non-null int64
5A 景区           31 non-null int64
rain            31 non-null object
TopMountains    31 non-null int64
dtypes: int64(6), object(3)
memory usage: 2.3+ KB


結論：共有31個行政區

In [72]:
 #展示盒狀圖的資訊
 df.describe()

Area(km2)  GDP(CNY 十亿)         人口(千人)      211大学     5A 景区  \
count  3.100000e+01     31.000000      31.000000  31.000000  31.00000   
mean   3.057437e+05   2759.580645   44366.193548   3.709677   8.00000   
std    3.779488e+05   2221.345325   28301.595415   4.947683   4.50925   
min    8.359000e+03    131.000000    3273.000000   1.000000   2.00000   
25%    1.152755e+05   1513.000000   24665.500000   1.000000   5.00000   
50%    1.760990e+05   2082.000000   38056.000000   2.000000   8.00000   
75%    3.103960e+05   3527.500000   60190.500000   4.000000  11.00000   
max    1.631585e+06   8988.000000  109240.000000  26.000000  23.00000   

       TopMountains  
count     31.000000  
mean    3185.548387  
std     2284.119405  
min       99.000000  
25%     1884.500000  
50%     2365.000000  
75%     3330.500000  
max     8844.000000

In [73]:
#檢視首欄標題
df.columns

Index(['行政区', '地区', 'Area(km2)', 'GDP(CNY 十亿)', '人口(千人)', '211大学', '5A 景区',
       'rain', 'TopMountains'],
      dtype='object')

In [74]:
#檢視首列標題
df.index

RangeIndex(start=0, stop=31, step=1)

In [75]:
#新資料表定義為，原資料表設定"行政區"做為首列標題，取前5列
df1 = df.set_index('行政区')
df1.head()

地区  Area(km2)  GDP(CNY 十亿)  人口(千人)  211大学  5A 景区   rain  TopMountains
行政区                                                                        
重庆   西南      82374          1950   30325      2      8  1,439          2797
云南   西南     383186          1653   47562      1      8  1,100          6740
西藏   西南    1202189           131    3273      1      2    400          8844
四川   西南     486116          3698   82330      5     12  1,100          7556
贵州   西南     176099          1354   35425      1      5  1,200          2900

In [76]:
#面積由小到大排序(邏輯：升冪排列正確)，取前5列
df1.sort_values('Area(km2)', ascending = True).head()

地区  Area(km2)  GDP(CNY 十亿)  人口(千人)  211大学  5A 景区   rain  TopMountains
行政区                                                                        
上海   华東       8359          3013   24175     10      3  1,124            99
天津   华北      11917          1860   15545      3      2    600          1052
北京   华北      16406          2800   21717     26      7    500          2303
海南   华南      35177           446    9140      1      6  1,639          1867
宁夏   西北      51893           345    6714      1      4    200          3556

以上結論：全中國面積最小的行政區為上海市，僅8,359平方公里

In [77]:
#面積由大到小排序(邏輯：降冪排列錯誤)，取前10列
df1.sort_values('Area(km2)',ascending = False).head(10)

地区  Area(km2)  GDP(CNY 十亿)  人口(千人)  211大学  5A 景区   rain  TopMountains
行政区                                                                        
新疆   西北    1631585          1092   23790      2     12    154          8611
西藏   西南    1202189           131    3273      1      2    400          8844
内蒙   华北    1145499          1610   25156      1      4    220          2365
青海   西北     696647           264    5909      1      3    380          6860
四川   西南     486116          3698   82330      5     12  1,100          7556
黑龙江  東北     452538          1620   38056      4      5    500          2036
甘肃   西北     425889           768   26050      1      4    300          5798
云南   西南     383186          1653   47562      1      8  1,100          6740
广西   华南     237606          2040   48170      1      5  1,600          2141
湖南   华中     211836          3459   68025      4      8  1,500          2099

結論：全中國面積最大的行政區為新疆區，達163萬1585平方公里

In [78]:
#人口由大到小排序(邏輯：升序排列錯誤)，取前10列
df1.sort_values('人口(千人)', ascending = False).head(10)

地区  Area(km2)  GDP(CNY 十亿)  人口(千人)  211大学  5A 景区   rain  TopMountains
行政区                                                                        
广东   华南     179656          8988  109240      4     12  1,800          1902
山东   华東     157901          7268   98969      3     11    710          1545
河南   华中     165664          4499   95062      1     13    700          2414
四川   西南     486116          3698   82330      5     12  1,100          7556
江苏   华東     106600          8590   79875     11     23  1,000           625
河北   华北     188545          3596   74475      1      8    600          2882
湖南   华中     211836          3459   68025      4      8  1,500          2099
安徽   华東     140140          2752   61696      3     11  1,200          1860
湖北   华中     185937          3652   58685      7     11  1,200          3105
浙江   华東     105506          5177   55645      1     16  1,400          1929

以上結論：全中國人口最多的省份為廣東省，達1億0924萬人

In [79]:
#人口由小到大排序(邏輯：升序排列正確)，取前10列
df1.sort_values('人口(千人)', ascending = True).head(10)

地区  Area(km2)  GDP(CNY 十亿)  人口(千人)  211大学  5A 景区   rain  TopMountains
行政区                                                                        
西藏   西南    1202189           131    3273      1      2    400          8844
青海   西北     696647           264    5909      1      3    380          6860
宁夏   西北      51893           345    6714      1      4    200          3556
海南   华南      35177           446    9140      1      6  1,639          1867
天津   华北      11917          1860   15545      3      2    600          1052
北京   华北      16406          2800   21717     26      7    500          2303
新疆   西北    1631585          1092   23790      2     12    154          8611
上海   华東       8359          3013   24175     10      3  1,124            99
内蒙   华北    1145499          1610   25156      1      4    220          2365
甘肃   西北     425889           768   26050      1      4    300          5798

以上結論：全中國人口最少的省份為新疆自治區，僅327萬人

In [80]:
#211大學由大到小排序(邏輯：升序排列錯誤)，取前3列
df1.sort_values('211大学', ascending = False).head(3)

地区  Area(km2)  GDP(CNY 十亿)  人口(千人)  211大学  5A 景区   rain  TopMountains
行政区                                                                        
北京   华北      16406          2800   21717     26      7    500          2303
江苏   华東     106600          8590   79875     11     23  1,000           625
上海   华東       8359          3013   24175     10      3  1,124            99

In [81]:
#211大學由小到大排序(邏輯：升序排列正確)，取前3列
df1.sort_values('211大学', ascending = True).head(3)

地区  Area(km2)  GDP(CNY 十亿)  人口(千人)  211大学  5A 景区   rain  TopMountains
行政区                                                                        
内蒙   华北    1145499          1610   25156      1      4    220          2365
云南   西南     383186          1653   47562      1      8  1,100          6740
西藏   西南    1202189           131    3273      1      2    400          8844

In [82]:
#計算全中國有多少間211大學
df1['211大学'].sum()

115

以上結論：211大學是為中國「面向21世紀、重點建設100所左右的高等學校和一批重點學科」之建設工程，全中國共有115間211大學，其中以北京市26間最多，可間接推論：因北京市擁有大量資源，才有辦法大量建設頂尖的大學；反之，內蒙、雲南、西藏等行政區的資源與資金相對都市少，因此重點大學數量也稀少，都各只有一所211大學。

### 用圖表呈現資料

In [83]:
#對GDP做長條圖，由小排到大
df1['GDP(CNY 十亿)'].sort_values().plot(kind='bar')

<Figure size 432x288 with 1 Axes>

In [84]:
#對GDP做圓餅圖，由小排到大
df1['GDP(CNY 十亿)'].sort_values().plot(kind='pie')

<Figure size 432x288 with 1 Axes>

以上結論：單單廣東省、江蘇省、山東省就佔了全中國近四分之一的GDP總額

In [85]:
#對人口做長條圖，由小排到大
df1['人口(千人)'].sort_values().plot(kind='bar')

<Figure size 432x288 with 1 Axes>

In [86]:
#對人口做圓餅圖，排序
df1['人口(千人)'].sort_values().plot(kind='pie')

<Figure size 432x288 with 1 Axes>

In [87]:
#對人口做盒狀圖
df1['人口(千人)'].plot(kind='box')

<Figure size 432x288 with 1 Axes>

以上結論：中國人口中位數約4000萬人，第3四分位數(Q3)約為6000萬人，第1四分位數(Q1)約為2000萬人

In [88]:
#二維數據圖，觀察人口與GDP的關係
df1.plot.scatter(x='人口(千人)', y='GDP(CNY 十亿)')

<Figure size 432x288 with 1 Axes>

以上結論：人口與GDP呈現正相關

## 欄位的處理

In [89]:
df1.head()

地区  Area(km2)  GDP(CNY 十亿)  人口(千人)  211大学  5A 景区   rain  TopMountains
行政区                                                                        
重庆   西南      82374          1950   30325      2      8  1,439          2797
云南   西南     383186          1653   47562      1      8  1,100          6740
西藏   西南    1202189           131    3273      1      2    400          8844
四川   西南     486116          3698   82330      5     12  1,100          7556
贵州   西南     176099          1354   35425      1      5  1,200          2900

In [90]:
#建新的欄位"人口密度"，降冪排序，取前五列
df1['人口密度']=df1['人口(千人)']*1000/df1['Area(km2)']
df1.sort_values('人口密度', ascending=False).head()

地区  Area(km2)  GDP(CNY 十亿)  人口(千人)  211大学  5A 景区   rain  TopMountains  \
行政区                                                                           
上海   华東       8359          3013   24175     10      3  1,124            99   
北京   华北      16406          2800   21717     26      7    500          2303   
天津   华北      11917          1860   15545      3      2    600          1052   
江苏   华東     106600          8590   79875     11     23  1,000           625   
山东   华東     157901          7268   98969      3     11    710          1545   

            人口密度  
行政区               
上海   2892.092356  
北京   1323.723028  
天津   1304.439037  
江苏    749.296435  
山东    626.778804

以上結論：全中國人口密度最高的行政區為上海市，每平方公里達2892人

In [91]:
#建新的欄位"人口密度"，降冪排序，取前五列
df1['人口密度']=df1['人口(千人)']*1000/df1['Area(km2)']
df1.sort_values('人口密度', ascending=False).tail()

地区  Area(km2)  GDP(CNY 十亿)  人口(千人)  211大学  5A 景区 rain  TopMountains  \
行政区                                                                         
甘肃   西北     425889           768   26050      1      4  300          5798   
内蒙   华北    1145499          1610   25156      1      4  220          2365   
新疆   西北    1631585          1092   23790      2     12  154          8611   
青海   西北     696647           264    5909      1      3  380          6860   
西藏   西南    1202189           131    3273      1      2  400          8844   

          人口密度  
行政区             
甘肃   61.166172  
内蒙   21.960735  
新疆   14.580914  
青海    8.482058  
西藏    2.722534

以上結論：全中國人口密度最低的行政區為西藏自治區，每平方公里僅2.7人

In [92]:
#建新的欄位"人均GDP(美元)"，2019年7月3日US$1 = RMB$6.88，降冪排序，取前五列
df1['人均GDP(美元)']=(df1['GDP(CNY 十亿)']*1000000000)/(df1['人口(千人)']*1000*6.88)
df1.sort_values('人均GDP(美元)', ascending = False).head()

地区  Area(km2)  GDP(CNY 十亿)  人口(千人)  211大学  5A 景区   rain  TopMountains  \
行政区                                                                           
北京   华北      16406          2800   21717     26      7    500          2303   
上海   华東       8359          3013   24175     10      3  1,124            99   
天津   华北      11917          1860   15545      3      2    600          1052   
江苏   华東     106600          8590   79875     11     23  1,000           625   
浙江   华東     105506          5177   55645      1     16  1,400          1929   

            人口密度     人均GDP(美元)  
行政区                             
北京   1323.723028  18740.007560  
上海   2892.092356  18115.244944  
天津   1304.439037  17391.369393  
江苏    749.296435  15631.255232  
浙江    527.410763  13522.705189

以上結論：北京市人均GDP為全中國最高，為US$18,740/人

In [93]:
#建新的欄位"人均GDP(美元)"，2019年7月3日US$1 = RMB$6.88，降冪排序，取前五列
df1['人均GDP(美元)']=(df1['GDP(CNY 十亿)']*1000000000)/(df1['人口(千人)']*1000*6.88)
df1.sort_values('人均GDP(美元)', ascending = False).tail()

地区  Area(km2)  GDP(CNY 十亿)  人口(千人)  211大学  5A 景区   rain  TopMountains  \
行政区                                                                           
山西   华北     156698          1497   36728      1      7    520          3058   
西藏   西南    1202189           131    3273      1      2    400          8844   
贵州   西南     176099          1354   35425      1      5  1,200          2900   
云南   西南     383186          1653   47562      1      8  1,100          6740   
甘肃   西北     425889           768   26050      1      4    300          5798   

           人口密度    人均GDP(美元)  
行政区                           
山西   234.387165  5924.286901  
西藏     2.722534  5817.506164  
贵州   201.165254  5555.464378  
云南   124.122489  5051.545938  
甘肃    61.166172  4285.140383

以上結論：甘肅省人均GDP為全中國最低，為US$4,285/人

In [94]:
#先複製資料表
#新的'rain'欄位為原先'rain'欄位應用/套用一個轉換所得來的
#其中lambda x為暫時的虛擬函式，它是由原先有分號的值，透過'replace'轉換成沒有分號的值所得來的
df2 = df1
df2['rain']=df2['rain'].apply(lambda x: int(x.replace(',','')))
df2.head()

地区  Area(km2)  GDP(CNY 十亿)  人口(千人)  211大学  5A 景区  rain  TopMountains  \
行政区                                                                          
重庆   西南      82374          1950   30325      2      8  1439          2797   
云南   西南     383186          1653   47562      1      8  1100          6740   
西藏   西南    1202189           131    3273      1      2   400          8844   
四川   西南     486116          3698   82330      5     12  1100          7556   
贵州   西南     176099          1354   35425      1      5  1200          2900   

           人口密度    人均GDP(美元)  
行政区                           
重庆   368.138005  9346.421519  
云南   124.122489  5051.545938  
西藏     2.722534  5817.506164  
四川   169.362868  6528.604397  
贵州   201.165254  5555.464378

In [95]:
#篩選單一條件
df1[(df1['人口(千人)']>40000)].head()

地区  Area(km2)  GDP(CNY 十亿)  人口(千人)  211大学  5A 景区  rain  TopMountains  \
行政区                                                                          
云南   西南     383186          1653   47562      1      8  1100          6740   
四川   西南     486116          3698   82330      5     12  1100          7556   
湖南   华中     211836          3459   68025      4      8  1500          2099   
湖北   华中     185937          3652   58685      7     11  1200          3105   
河南   华中     165664          4499   95062      1     13   700          2414   

           人口密度    人均GDP(美元)  
行政区                           
云南   124.122489  5051.545938  
四川   169.362868  6528.604397  
湖南   321.121056  7390.836132  
湖北   315.617655  9045.138510  
河南   573.824126  6878.925529

In [96]:
#篩選多種條件
df1[(df1['人口(千人)']>40000) &(df1['Area(km2)']>150000)].head()

地区  Area(km2)  GDP(CNY 十亿)  人口(千人)  211大学  5A 景区  rain  TopMountains  \
行政区                                                                          
云南   西南     383186          1653   47562      1      8  1100          6740   
四川   西南     486116          3698   82330      5     12  1100          7556   
湖南   华中     211836          3459   68025      4      8  1500          2099   
湖北   华中     185937          3652   58685      7     11  1200          3105   
河南   华中     165664          4499   95062      1     13   700          2414   

           人口密度    人均GDP(美元)  
行政区                           
云南   124.122489  5051.545938  
四川   169.362868  6528.604397  
湖南   321.121056  7390.836132  
湖北   315.617655  9045.138510  
河南   573.824126  6878.925529

###  資料群組化


In [97]:
#將地區群組化，計算每個地區有幾個省
df2a = df1.groupby('地区').count()
df2a

Area(km2)  GDP(CNY 十亿)  人口(千人)  211大学  5A 景区  rain  TopMountains  人口密度  \
地区                                                                            
华中          3             3       3      3      3     3             3     3   
华北          5             5       5      5      5     5             5     5   
华南          3             3       3      3      3     3             3     3   
华東          7             7       7      7      7     7             7     7   
東北          3             3       3      3      3     3             3     3   
西北          5             5       5      5      5     5             5     5   
西南          5             5       5      5      5     5             5     5   

    人均GDP(美元)  
地区             
华中          3  
华北          5  
华南          3  
华東          7  
東北          3  
西北          5  
西南          5

In [98]:
#按地區群組化，其餘欄位取平均數(也可取最大值、最小值)
df2b = df1.groupby('地区').mean()
df2b

Area(km2)  GDP(CNY 十亿)        人口(千人)     211大学      5A 景区  \
地区                                                                   
华中  187812.333333   3870.000000  73924.000000  4.000000  10.666667   
华北  303813.000000   2272.600000  34724.200000  6.400000   5.600000   
华南  150813.000000   3824.666667  55516.666667  2.000000   7.666667   
华東  115627.571429   4587.428571  57816.428571  4.428571  11.857143   
東北  264039.666667   1847.666667  36429.666667  3.666667   5.333333   
西北  602327.600000    931.800000  20098.000000  2.600000   6.200000   
西南  465992.800000   1757.200000  39783.000000  2.000000   7.000000   

           rain  TopMountains        人口密度     人均GDP(美元)  
地区                                                       
华中  1133.333333   2539.333333  403.520946   7771.633391  
华北   488.000000   2332.000000  655.901701  11675.240193  
华南  1679.666667   1970.000000  356.870126   8402.328648  
华東  1233.428571   1438.857143  831.607368  11887.006264  
東北   616.666667   2021.000000  174.254219   7410.995966  
西北   326.800000   5718.400000   79.709079   6658.047680  
西南  1047.800000   5767.400000  173.102230   6459.908479

In [99]:
#平均人口按順序做長條圖
df2b['人口(千人)'].sort_values().plot(kind='bar')

<Figure size 432x288 with 1 Axes>

In [100]:
#平均最高峰的高度按順序做長條途
df2b['TopMountains'].sort_values().plot(kind='bar')

<Figure size 432x288 with 1 Axes>

In [101]:
#按地區群組化，其他欄位加總
df2c=df1.groupby('地区').sum()
df2c.head()

Area(km2)  GDP(CNY 十亿)  人口(千人)  211大学  5A 景区  rain  TopMountains  \
地区                                                                      
华中     563437         11610  221772     12     32  3400          7618   
华北    1519065         11363  173621     32     28  2440         11660   
华南     452439         11474  166550      6     23  5039          5910   
华東     809393         32112  404715     31     83  8634         10072   
東北     792119          5543  109289     11     16  1850          6063   

           人口密度     人均GDP(美元)  
地区                             
华中  1210.562837  23314.900172  
华北  3279.508506  58376.200963  
华南  1070.610378  25206.985944  
华東  5821.251575  83209.043846  
東北   522.762658  22232.987898

In [102]:
#人口數與大學數量的關係圖
df2c.plot.scatter(x='人口(千人)',y='211大学')

<Figure size 432x288 with 1 Axes>

In [36]:
#各地區大學數量按排序作圓餅圖
df2c['211大学'].sort_values().plot(kind='pie')

<Figure size 432x288 with 1 Axes>

### 對欄位分級

In [106]:
#複製一份表格
df3 = df1
df3.head()

地区  Area(km2)  GDP(CNY 十亿)  人口(千人)  211大学  5A 景区  rain  TopMountains  \
行政区                                                                          
重庆   西南      82374          1950   30325      2      8  1439          2797   
云南   西南     383186          1653   47562      1      8  1100          6740   
西藏   西南    1202189           131    3273      1      2   400          8844   
四川   西南     486116          3698   82330      5     12  1100          7556   
贵州   西南     176099          1354   35425      1      5  1200          2900   

           人口密度    人均GDP(美元) 人口密度等級  
行政区                                  
重庆   368.138005  9346.421519     擁擠  
云南   124.122489  5051.545938     普通  
西藏     2.722534  5817.506164     稀疏  
四川   169.362868  6528.604397     普通  
贵州   201.165254  5555.464378     擁擠

In [107]:
#對人口密度分四個等級：極擁擠(500人/km2以上)，擁擠(200人/km2以上)，普通(50至200人/km2)，稀疏(50人/km2以下)
df3['人口密度等級']=df3['人口密度'].apply(lambda x: '極擁擠' if (x>500) else('擁擠' if (x>200) else('普通' if (x>50) else'稀疏')))
df3

地区  Area(km2)  GDP(CNY 十亿)  人口(千人)  211大学  5A 景区  rain  TopMountains  \
行政区                                                                          
重庆   西南      82374          1950   30325      2      8  1439          2797   
云南   西南     383186          1653   47562      1      8  1100          6740   
西藏   西南    1202189           131    3273      1      2   400          8844   
四川   西南     486116          3698   82330      5     12  1100          7556   
贵州   西南     176099          1354   35425      1      5  1200          2900   
新疆   西北    1631585          1092   23790      2     12   154          8611   
陕西   西北     205624          2190   38027      8      8   600          3767   
青海   西北     696647           264    5909      1      3   380          6860   
宁夏   西北      51893           345    6714      1      4   200          3556   
甘肃   西北     425889           768   26050      1      4   300          5798   
湖南   华中     211836          3459   68025      4      8  1500          2099   
湖北   华中     185937          3652   58685      7     11  1200          3105   
河南   华中     165664          4499   95062      1     13   700          2414   
海南   华南      35177           446    9140      1      6  1639          1867   
广西   华南     237606          2040   48170      1      5  1600          2141   
广东   华南     179656          8988  109240      4     12  1800          1902   
浙江   华東     105506          5177   55645      1     16  1400          1929   
上海   华東       8359          3013   24175     10      3  1124            99   
山东   华東     157901          7268   98969      3     11   710          1545   
江西   华東     166936          2082   45790      1     10  1600          2158   
江苏   华東     106600          8590   79875     11     23  1000           625   
福建   华東     123951          3230   38565      2      9  1600          1856   
安徽   华東     140140          2752   61696      3     11  1200          1860   
天津   华北      11917          1860   15545      3      2   600          1052   
山西   华北     156698          1497   36728      1      7   520          3058   
内蒙   华北    1145499          1610   25156      1      4   220          2365   
河北   华北     188545          3596   74475      1      8   600          2882   
北京   华北      16406          2800   21717     26      7   500          2303   
辽宁   東北     148379          2394   43801      4      5   800          1336   
吉林   東北     191202          1529   27432      3      6   550          2691   
黑龙江  東北     452538          1620   38056      4      5   500          2036   

            人口密度     人均GDP(美元) 人口密度等級  
行政区                                    
重庆    368.138005   9346.421519     擁擠  
云南    124.122489   5051.545938     普通  
西藏      2.722534   5817.506164     稀疏  
四川    169.362868   6528.604397     普通  
贵州    201.165254   5555.464378     擁擠  
新疆     14.580914   6671.749905     稀疏  
陕西    184.934638   8370.735359     普通  
青海      8.482058   6493.838725     稀疏  
宁夏    129.381612   7468.774030     普通  
甘肃     61.166172   4285.140383     普通  
湖南    321.121056   7390.836132     擁擠  
湖北    315.617655   9045.138510     擁擠  
河南    573.824126   6878.925529    極擁擠  
海南    259.828865   7092.514376     擁擠  
广西    202.730571   6155.524765     擁擠  
广东    608.050942  11958.946804    極擁擠  
浙江    527.410763  13522.705189    極擁擠  
上海   2892.092356  18115.244944    極擁擠  
山东    626.778804  10674.002454    極擁擠  
江西    274.296736   6608.785304     擁擠  
江苏    749.296435  15631.255232    極擁擠  
福建    311.131011  12173.648235     擁擠  
安徽    440.245469   6483.402490     擁擠  
天津   1304.439037  17391.369393    極擁擠  
山西    234.387165   5924.286901     擁擠  
内蒙     21.960735   9302.418028     稀疏  
河北    394.998541   7018.119082     擁擠  
北京   1323.723028  18740.007560    極擁擠  
辽宁    295.196760   7944.227672     擁擠  
吉林    143.471303   8101.427971     普通  
黑龙江    84.094595   6187.332255     普通

In [108]:
#計算各種人口密度等級的行政區分別有多少
df3a = df3
df3a = df3.groupby('人口密度等級').count()
df3a

地区  Area(km2)  GDP(CNY 十亿)  人口(千人)  211大学  5A 景区  rain  TopMountains  \
人口密度等級                                                                          
擁擠      12         12            12      12     12     12    12            12   
普通       7          7             7       7      7      7     7             7   
極擁擠      8          8             8       8      8      8     8             8   
稀疏       4          4             4       4      4      4     4             4   

        人口密度  人均GDP(美元)  
人口密度等級                   
擁擠        12         12  
普通         7          7  
極擁擠        8          8  
稀疏         4          4

In [109]:
#用圓餅圖呈現各種人口密度等級的占比
df3a['地区'].sort_values().plot(kind='pie')

<Figure size 432x288 with 1 Axes>

以上結論：單單極擁擠(500人/km2以上)與擁擠(200人/km2以上)的省份就佔了全中國一半以上